### PR TITLE
Add a GitLab CI pipeline

### PR DESCRIPTION
**Pull request by ghost**
This is our 2nd GitLab pipeline, very similar to the one for summon.

https://github.com/cyberark/summon/pull/72

To get this working I had to parameterize the test runner to be able to run against OSS on gitlab, and OSS and Enterprise on Jenkins.

### CODE REVIEW

**Comment by ghost** [comment]
This requires no version bump. Please squash merge.

File: Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ MAINTAINER Conjur Inc
 RUN apt-get update && apt-get install -y jq less 
 RUN go get -u github.com/jstemmer/go-junit-report
 RUN go get -u github.com/golang/dep/cmd/dep
-RUN go get github.com/smartystreets/goconvey
+RUN go get github.com/playscale/goconvey
```
Comment (apotterri) [inline Dockerfile:7]: Why the change here?

File: build.sh [hunk 2]
```diff
@@ -15,11 +14,14 @@ PLATFORMS=(
   # 'windows:386'
   'windows:amd64'
 )
+PLATFORMS="${1:-${DEFAULT_PLATFORMS[@]}}"  # override this with a positional argument, like 'linux:amd64'
+
+OUTPUT_DIR='output'
 
-echo "Creating summon-conjur binaries in output/"
+echo "Creating summon binaries in $OUTPUT_DIR/"
```
Comment (apotterri) [inline build.sh]: This should still be summon-conjur, shouldn't it?

File: functions.sh [hunk 3]
```diff
@@ -1,30 +1,48 @@
-startConjur() {
-  # Start a local Conjur to test against
-  docker-compose pull postgres conjur cuke-master
-  docker-compose up -d conjur cuke-master
+#!/bin/bash
```
Comment (apotterri) [inline functions.sh]: It doesn't make sense to have a shebang here -- this file is only ever sourced, not run.

File: package_enterprise_test.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+	ApplianceURL_V4 := os.Getenv("CONJUR_V4_APPLIANCE_URL")
+	SSLCert_V4 := os.Getenv("CONJUR_V4_SSL_CERTIFICATE")
+	APIKey_V4 := os.Getenv("CONJUR_V4_AUTHN_API_KEY")
+
+	Path := os.Getenv("PATH")
+
+	Convey("Given a compiled summon-conjur package", t, func() {
+		Convey("Given a v4 appliance", func() {
+			Convey("Given valid configuration", func() {
```
Comment (apotterri) [inline package_enterprise_test.go]: The top two levels of nesting here don't seem like they add much clarity to what's being tested. A compiled summon-conjur package and a v4 appliance are both absolute requirements for this set of tests.

Can we get rid of them?

File: test-entrypoint.sh [hunk 5]
```diff
@@ -1,13 +1,14 @@
-#!/bin/bash
+#!/bin/bash -e
```
Comment (apotterri) [inline test-entrypoint.sh]: We may want `pipefail` set here, too, to make sure we exit non-zero if `go test` in the pipeline below fails.

**Review by apotterri — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (ghost) [reply Dockerfile:7]: This is because I wanted to use `-tags` for enabling tests, oss vs enterprise. smartystreets/goconvey doesn't support tags, doesn't look like there has been any development on the repo since February 2018.

I found the playscale repo via this PR, looks like they are keeping goconvey more up to date, with support for tags

https://github.com/smartystreets/goconvey/pull/458

File: package_enterprise_test.go (hunk 4 above)
Comment (ghost) [reply package_enterprise_test.go]: yep this was just a leftover, i'll resolve it.

File: Dockerfile (hunk 1 above)
Comment (apotterri) [reply Dockerfile:7]: Ok, good to know.

File: package_enterprise_test.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	Account := os.Getenv("CONJUR_ACCOUNT")
+	Login := os.Getenv("CONJUR_AUTHN_LOGIN")
+
+	ApplianceURL_V4 := os.Getenv("CONJUR_V4_APPLIANCE_URL")
+	SSLCert_V4 := os.Getenv("CONJUR_V4_SSL_CERTIFICATE")
+	APIKey_V4 := os.Getenv("CONJUR_V4_AUTHN_API_KEY")
+
+	Path := os.Getenv("PATH")
+
+	Convey("Given a compiled summon-conjur package", t, func() {
```
Comment (apotterri) [inline package_enterprise_test.go]: I still think this one could go away, too. Unless, somehow, we're going to try to run tests when there isn't a "compiled summon-conjur package"?

Comment (ghost) [reply package_enterprise_test.go]: Sure I'll remove it - we're explicitly creating the package with `build` before the tests.

**Review by apotterri — APPROVED** [review]
